pkg/apis/resource/validation/validation.go

```diff
@@ -738,16 +742,37 @@ func validateRawExtension(rawExtension runtime.RawExtension, fldPath *field.Path
 	return allErrs
 }
 
+const interfaceNameMaxLength int = 256
+const hardwareAddressMaxLength int = 128
+
 func validateNetworkDeviceData(networkDeviceData *resource.NetworkDeviceData, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 	if networkDeviceData == nil {
 		return allErrs
 	}
-	allErrs = append(allErrs, validateSlice(networkDeviceData.Addresses, -1,
+
+	if len(networkDeviceData.InterfaceName) > interfaceNameMaxLength {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("InterfaceName"), networkDeviceData.InterfaceName, validation.MaxLenError(interfaceNameMaxLength)))
```

Comment:
use `field.TooLong(fldPath.Child("interfaceName"), "" /* unused */, interfaceNameMaxLength)`

Note that the "" arg is unused and the field-path uses JSON-style names :)

Reply:
`field.TooLong` will return `must have at most %d bytes`
`MaxLenError` returns `must be no more than %d characters`.
Otherwise there is `TooLongMaxLength` returning `may not be longer than %d`.

Reply:
`TooLongMaxLength` is going away.

"bytes" is actually correct - `len(string)` returns bytes, not characters.

Reply:
Fixed, thank you

Reply:
TooLongMaxLength is going away - please use `field.TooLong(fldPath.Child("interfaceName"), "" /* unused */, interfaceNameMaxLength)`

Reply:
ho ok, sorry, I miss understood the previous message, now I set `field.TooLong`